src/main/java/cn/nukkit/network/protocol/DataPacket.java

```diff
@@ -63,18 +68,18 @@ public DataPacket clone() {
     }
 
     public BatchPacket compress() {
-        return compress(Server.getInstance().networkCompressionLevel);
+        return compress(Server.getInstance().networkCompressionLevel, ProtocolInfo.CURRENT_PROTOCOL);
```

Comment:
No point having this method here if backward compatibility for everything else here is broken